org.mwc.debrief.core/src/org/mwc/debrief/core/ui/ImportNarrativeHelper.java

```diff
@@ -0,0 +1,64 @@
+package org.mwc.debrief.core.ui;
+
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.widgets.Display;
+import org.mwc.cmap.core.CorePlugin;
+
+import Debrief.ReaderWriter.Word.ImportNarrativeDocument.ImportNarrativeEnum;
+import Debrief.ReaderWriter.Word.ImportNarrativeDocument.TrimNarrativeHelper;
+
+/**
+ * Helper class to pop up dialog to offer choice to analyst
+ * to import all data or loaded tracks 
+ * @author Ayesha <ayesha.ma@gmail.com>
+ *
+ */
+public class ImportNarrativeHelper implements TrimNarrativeHelper
+{
+
+  private static final String PREF_DEF_NARRATIVE_CHOICE = "defaultNarrativeEntryChoice";
+  @Override
+  public ImportNarrativeEnum findWhatToImport()
+  {
+    final Display targetDisplay;
+    String defaultChoice = CorePlugin.getDefault().getPreference(PREF_DEF_NARRATIVE_CHOICE);

```

Comment:
@ayeshare - users also need a way to clear the "re-use next time" choice.  

So, we need a checkbox in _Debrief specific settings_.  It should be titled "Re-use existing choice for trimming imported narratives".  Once the user has selected a preference, the box should get ticked.  But, when the box gets cleared by the user, the popup dialog should start appearing again.

I guess the value of the checkbox is controlled by the presence of the default choice preference value. If one is there, the box gets ticked.  If one isn't assigned, the box is clear.  

Unticking the box would clear the default choice.

Reply:
implemented the preference